luqezman/fullstack-developer-hiring-exercise
# Feat/register

## Description
Added new feature to create a new account

## Type of change
Please delete options that are not relevant.


## Migrations
YES

## Steps to Test or Reproduce

Make sure you are in the root of the project.

Installing database(make sure docker is running)

```sh
# go to database folder
cd database

# to generate database & run postgres
docker-compose up -d
```

Installing & running backend

```sh
# go to backend folder
cd backend

# install packages
npm install

# run migrations
npm run typeorm -- migration:run

# start backend
npm run start:dev

# if you want to inspect api go to http://localhost:3000/swagger
```

Installing & running frontend

```sh
# go to frontend folder
cd backend

# install packages
npm install

# start frontend
ng serve
```

# Running tests

```sh
# to create database test
docker exec -it local_db psql -U local_dev -c "create database local_db_test" local_db

# starting backend within testing database
cd backend && NODE_ENV=testing npm run start:dev

# running e2e frontend tests
cd frontend && ng e2e
```

# Tests output

```sh
End-to-end tests for full stack exercise
    √ should show required errors when form is not filled
[1364:3224:0629/124434.136:ERROR:device_event_log_impl.cc(208)] [12:44:34.136] Bluetooth: bluetooth_adapter_winrt.cc:1060 Getting Default Adapter failed.
    √ should show validation errors when form is filled incorrectly
    √ should signup when form is filled correctly
    √ should show duplicated errors type when displayname/email are already taken
    √ should login with valid username and password
    √ should give an error for missing form field
    √ should give an error for bad password
    √ should display user information in dashboard
    √ should log out
```

## Checklist:
